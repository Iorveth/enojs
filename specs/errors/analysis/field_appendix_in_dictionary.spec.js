const eno = require('../../../eno.js');

const input = `
languages:
eno = eno notation
| json
`.trim();

describe('analysis.fieldAppendInDictionary', () => {

  let error;

  beforeAll(() => {
    try {
      eno.parse(input);
    } catch(err) {
      error = err;
    }
  })

  it(`provides the correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
