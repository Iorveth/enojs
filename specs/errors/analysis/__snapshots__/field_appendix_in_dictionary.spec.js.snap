// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analysis.fieldAppendInDictionary provides correct selection metadata 1`] = `
Array [
  Array [
    2,
    0,
  ],
  Array [
    2,
    6,
  ],
]
`;

exports[`analysis.fieldAppendInDictionary provides the correct message 1`] = `
"Line 3 contains an appendix (to a field) inside a dictionary.

   Line | Content
 *    1 | languages:
 *    2 | eno = eno notation
 >    3 | | json
"
`;
